pkg/dashboard/http.go

```diff
@@ -0,0 +1,55 @@
+package dashboard
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"k8s.io/apimachinery/pkg/util/json"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// SetupRouterAndServeHTTP setups and starts the http server. If everything goes well, it never returns.
+func SetupRouterAndServeHTTP(ctx context.Context, cl client.Client) error {
+	router := routes(ctx, cl)
+	server := &http.Server{
+		Addr:              ":80",
+		ReadHeaderTimeout: 5 * time.Second,
+		Handler:           router,
+	}
+	return server.ListenAndServe()
+}
+
+func routes(ctx context.Context, cl client.Client) http.Handler {
+	router := mux.NewRouter().StrictSlash(true)
+
+	// Routes to serve
+	router.HandleFunc("/api/foreign_clusters", foreignClusters(ctx, cl))
+
+	return router
+}
+
+func foreignClusters(ctx context.Context, cl client.Client) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		clusters, err := getForeignClusters(ctx, cl)
+		if err != nil {
+			w.WriteHeader(http.StatusInternalServerError)
+			err := json.NewEncoder(w).Encode(ErrorResponse{
+				Message: "An error occurred during retrieving foreign clusters and metrics",
+				Status:  500,
+				Details: err.Error(),
+			})
+			if err != nil {
+				return
+			}
+			return
```

Comment:
```suggestion
			return
```

Either you print a warning, or could be simplified.